tracing-subscriber/src/registry/fmt_layer.rs

```diff
@@ -0,0 +1,197 @@
+use crate::{
+    fmt::{format, FormatEvent, FormatFields, MakeWriter},
+    layer::{Context, Layer},
+    registry::{Extensions, LookupMetadata, LookupSpan, Registry, SpanData, SpanRef},
+};
+use ansi_term::{Color, Style};
+use humantime;
+use std::{
+    any::type_name,
+    fmt::{self, Write as _},
+    io::{self, Write},
+    marker::PhantomData,
+    time::SystemTime,
+};
+use tracing_core::{
+    field::{Field, Visit},
+    span::{Attributes, Id},
+    Event, Level, Subscriber,
+};
+
+pub struct FmtLayer<
+    S = Registry,
+    N = format::DefaultFields,
+    E = format::Format<format::Full>,
+    W = fn() -> io::Stdout,
+> {
+    is_interested: Box<dyn Fn(&Event<'_>) -> bool + Send + Sync + 'static>,
+    inner: PhantomData<S>,
+    make_writer: W,
+    fmt_fields: N,
+    fmt_event: E,
+}
+
+pub struct FmtLayerBuilder<
+    S = Registry,
+    N = format::DefaultFields,
+    E = format::Format<format::Full>,
+    W = fn() -> io::Stdout,
+> {
+    fmt_fields: N,
+    fmt_event: E,
+    make_writer: W,
+    is_interested: Box<dyn Fn(&Event<'_>) -> bool + Send + Sync + 'static>,
+    inner: PhantomData<S>,
+}
+
+impl FmtLayer {
+    pub fn builder() -> FmtLayerBuilder {
+        FmtLayerBuilder::default()
+    }
+}
+
+impl<S, N, E, W> FmtLayerBuilder<S, N, E, W>
+where
+    S: Subscriber + for<'a> LookupSpan<'a> + LookupMetadata,
+    N: for<'writer> FormatFields<'writer> + 'static,
+    E: FormatEvent<N> + 'static,
+    W: MakeWriter + 'static,
+{
+    pub fn with_interest<F>(self, f: F) -> Self
```

Comment:
I don't care for the naming "interest", because that implies to me that this participates in the global callsite filter caching, which i don't think this does. Can we come up with a name (+ docs) that better communicate that this is a separate way of having individual layers skip events that _have_ been enabled?

Reply:
Agreed, yeah. This could be the start of the work in #302?